docs/api/url.md

```diff
@@ -0,0 +1,86 @@
+# 🔗 URL
+An API for working with HTTP URLs. 
+
+## Quick Start
+
+We can construct a url in two ways, either as an absolute url or a relative url.
+```typescript 
+let url : URL = new URL(`tezos://${my_contract.address}/entrypoint`);
+let url2 : URL = new URL('../entrypoint_2', url.href);
+```
+The constructor will raise and exception if the arguments cannot be parsed into a valid url.
+The [`URL.canParse()`](#canParse) method can be used to check if the arguments can be parsed correctly.
+```typescript
+if (URL.canParse(foo, bar)) {
+  let url = new URL(foo, bar)
+  console.log(url.href)
+} else {
+  // the URL cannot be parsed, take appropriate action.
+  console.error("Invalid URL")
+}
+```
+
+We may edit or construct a url by setting values for its properties.
+```typescript
+let url = new URL('tezos://domain/);
+url.pathname = "my_entrypoint"
+url.hostname = Ledger.selfAddress();
+url.hash = "my_fragment"
+console.log(url.href) // tezos://tz4../my_entrypoint#my_fagment
+```
+
+Search parameters may be read or edited through the [`URLSearchParams`](./url_search_params.md) API.
```

Comment:
Makes `.searchParams` explicit in the description associated w/ the example below:

```suggestion
The [`URLSearchParams`](./url_search_params.md) API may be used to build and manipulate search parameters. To get the search parameters from the URL, you may use the `.searchParams` instance property.
```